hloc/extractors/dog.py

```diff
@@ -94,18 +96,24 @@ def _forward(self, data):
             raise ValueError(f'Unknown descriptor: {self.conf["descriptor"]}')
 
         keypoints = torch.from_numpy(keypoints[:, :2])  # keep only x, y
+        scales = torch.from_numpy(scales)
+        oris = torch.from_numpy(oris)
         scores = torch.from_numpy(scores)
 
         if self.conf['max_keypoints'] != -1:
             # TODO: check that the scores from PyCOLMAP are 100% correct,
             # follow https://github.com/mihaidusmanu/pycolmap/issues/8
             indices = torch.topk(scores, self.conf['max_keypoints'])
             keypoints = keypoints[indices]
+            scales = scales[indices]
+            oris = oris[indices]
             scores = scores[indices]
             descriptors = descriptors[indices]
 
         return {
             'keypoints': keypoints[None],
+            'scales': scales[None],
```

Comment:
When the input image is downsized, don't you need to also modify the scale in `extract_features.py` to account for it?

Reply:
Good catch, I'll fix it!